adapt/engine.py

```diff
@@ -121,8 +164,18 @@ def register_intent_parser(self, intent_parser):
 
         :raises ValueError on invalid intent
         """
-        if hasattr(intent_parser, 'validate') and callable(intent_parser.validate):
-            self.intent_parsers.append(intent_parser)
+        if hasattr(
+                intent_parser,
+                'validate') and callable(
+                intent_parser.validate):
+            missing_entities = self.trie.checkForMissingEntites(
```

Comment:
I think logging here is fine; this is a relatively low throughput area of the code, so no perf impact.